packages/router-vite-plugin/src/eliminateUnreferencedIdentifiers.ts

```diff
@@ -106,35 +106,6 @@ export const eliminateUnreferencedIdentifiers = (
           handleObjectPattern(
             path.get('id') as NodePath<BabelTypes.ObjectPattern>,
           )
-        } else if (path.node.id.type === 'ObjectPattern') {
-          const pattern = path.get('id') as NodePath<BabelTypes.ObjectPattern>
-
-          const beforeCount = referencesRemovedInThisPass
-          const properties = pattern.get('properties')
-          properties.forEach((property) => {
-            const local = property.get(
-              property.node.type === 'ObjectProperty'
-                ? 'value'
-                : // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-                  property.node.type === 'RestElement'
-                  ? 'argument'
-                  : (function () {
-                      throw new Error('invariant')
-                    })(),
-            ) as NodePath<BabelTypes.Identifier>
-
-            if (shouldBeRemoved(local)) {
-              ++referencesRemovedInThisPass
-              property.remove()
-            }
-          })
-
-          if (
-            beforeCount !== referencesRemovedInThisPass &&
-            pattern.get('properties').length < 1
-          ) {
-            path.remove()
-          }
```

Comment:
this seems to be a leftover as `path.node.id.type === 'ObjectPattern'` is checked one line above, which means this should be dead code